code/config/lib/Loot/Config/Buildable.hs

```diff
@@ -0,0 +1,92 @@
+{- SPDX-License-Identifier: MPL-2.0 -}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE GADTs #-}
+
+module Loot.Config.Buildable
+  (
+  ) where
+
+import Data.Vinyl (Rec (RNil, (:&)))
+import Fmt (Buildable (..), Builder, blockMapF)
+import GHC.TypeLits (KnownSymbol, symbolVal)
+
+import Loot.Config.Record (ConfigKind, ConfigRec, Item (..), ItemKind,
+                           SumSelection, (::+), (::-), (:::), (::<))
+
+class OptionsToBuilderList (k :: ConfigKind) (is :: [ItemKind]) where
+  configToBuilderList :: ConfigRec k is -> [(Text, Builder)]
+
+instance OptionsToBuilderList (k :: ConfigKind) '[] where
+  configToBuilderList RNil = []
+
+undefinedBuilder :: Builder
+undefinedBuilder = build @Text "<undefined>"
+
+instance
+  forall l v is k
+  . ( KnownSymbol l
+    , Buildable v
+    , OptionsToBuilderList k is
+    )
+  => OptionsToBuilderList k ((l ::: v) ': is)
+  where
+    configToBuilderList (itemOption :& rest) =
+      let label = fromString $ symbolVal (Proxy :: Proxy l)
```

Comment:
One more suggestion: there are several uses of
```haskell
fromString $ symbolVal (Proxy :: Proxy l)
```
that could be extracted to a function.

However that function already exists, it's `getConfigKey` in `Loot.Config.Yaml`.

So I suggest moving `getConfigKey` into `Loot.Config.Record` and use it in this module and `Loot.Config.Yaml`.

Reply:
good!
I forgot about this function, thanks